subgraphs/beefy-finance/src/utils/metrics.ts

```diff
@@ -0,0 +1,300 @@
+}
+
+export function getUniqueUsers(
+  protocol: YieldAggregator,
+  timeframe: BigInt[]
+): BigInt {
+  let vault: Vault | null,
+    deposit: Deposit | null,
+    withdraw: Withdraw | null,
+    user: string;
+  let users: string[] = [];
+  for (let i = 0; i < protocol.vaults.length; i++) {
+    vault = Vault.load(protocol.vaults[i]); //already initialized if are in vaults field
+    if (vault) {
+      for (let j = 0; j < vault.deposits.length; j++) {
+        deposit = Deposit.load(vault.deposits[j]);
+        if (
+          deposit &&
+          deposit.timestamp > timeframe[0] &&
+          deposit.timestamp <= timeframe[1]
+        ) {
+          user = deposit.from;
+          if (!users.includes(user)) {
+            users.push(user);
+          }
+        }
+      }
+      for (let k = 0; k < vault.withdraws.length; k++) {
+        withdraw = Withdraw.load(vault.withdraws[k]);
+        if (
+          withdraw &&
+          withdraw.timestamp > timeframe[0] &&
+          withdraw.timestamp <= timeframe[1]
+        ) {
+          user = withdraw.from;
+          if (!users.includes(user)) {
+            users.push(user);
+          }
+        }
+      }
+    }
+  }
+  return BigInt.fromI32(users.length);
+}
+
+function getDepositCount(
+  protocol: YieldAggregator,
+  timeframe: BigInt[]
+): BigInt {
+  let vault: Vault | null, deposit: Deposit | null;
+  let count = BIGINT_ZERO;
+  for (let i = 0; i < protocol.vaults.length; i++) {
+    vault = Vault.load(protocol.vaults[i]); //already initialized if are in vaults field
+    if (vault == null) {
+      continue;
+    } else {
+      for (let j = 0; j < vault.deposits.length; j++) {
+        deposit = Deposit.load(vault.deposits[j]);
+        if (deposit == null) {
+          continue;
+        } else if (
+          deposit.timestamp >= timeframe[0] &&
+          deposit.timestamp <= timeframe[1]
+        ) {
+          count = count.plus(BigInt.fromI32(1));
+        }
+      }
+    }
+  }
+  return count;
+}
+
+function getWithdrawCount(
+  protocol: YieldAggregator,
+  timeframe: BigInt[]
+): BigInt {
+  let vault: Vault | null, withdraw: Withdraw | null;
+  let count = BIGINT_ZERO;
+  for (let i = 0; i < protocol.vaults.length; i++) {
+    vault = Vault.load(protocol.vaults[i]); //already initialized if are in vaults field
+    if (vault == null) {
+      continue;
+    } else {
+      for (let j = 0; j < vault.withdraws.length; j++) {
+        withdraw = Withdraw.load(vault.withdraws[j]);
+        if (withdraw == null) {
+          continue;
+        } else if (
+          withdraw.timestamp >= timeframe[0] &&
+          withdraw.timestamp <= timeframe[1]
+        ) {
+          count = count.plus(BigInt.fromI32(1));
+        }
+      }
+    }
+  }
+  return count;
+}
+
+function getTransactionCount(
+  protocol: YieldAggregator,
+  timeframe: BigInt[]
+): BigInt {
+  return getDepositCount(protocol, timeframe).plus(
+    getWithdrawCount(protocol, timeframe)
+  );
+}
```

Comment:
This should be handled at the transaction level. Example: [Convex Subgraph](https://github.com/messari/subgraphs/blob/master/subgraphs/convex/src/modules/Metric.ts#L14-L64)